### PR TITLE
feat(core): new "is journal" page property

### PR DESCRIPTION
**Pull request by CatsJuice**
close AF-1450, AF-1451, AF-14552

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#8525** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8525?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/modules/journal/store/journal.ts [hunk 1]
```diff
@@ -11,6 +11,17 @@ export class JournalStore extends Store {
...
   }
 
+  allJournalDates$ = LiveData.computed(get => {
+    return new Set(
+      get(this.docsService.list.docs$)
+        .filter(doc => {
+          const journal = get(doc.properties$.selector(p => p.journal));
+          return !!journal && isJournalString(journal);
+        })
+        .map(doc => doc.properties$.value.journal)
```
Comment (pengx17) [inline packages/frontend/core/src/modules/journal/store/journal.ts]: get(doc.properties$).journal?

Comment (CatsJuice) [reply packages/frontend/core/src/modules/journal/store/journal.ts]: Not really sure, I think that should already be observed inside `filter`. It seems to heavy here.

File: packages/frontend/core/src/desktop/pages/workspace/detail-page/tabs/journal.tsx [hunk 2]
```diff
@@ -297,6 +322,12 @@ const ConflictList = ({
     },
     [setTrashModal]
   );
+  const handleRemoveJournalMark = useCallback(
+    (docId: string) => {
+      journalService.removeJournalDate(docId);
```
Comment (pengx17) [inline packages/frontend/core/src/desktop/pages/workspace/detail-page/tabs/journal.tsx:329]: worth adding a telemetry event

**Review by pengx17 — APPROVED** [review]
I think we need to add tests for the duplicated journal logic. However how to add duplicated journal is a question ... Perhaps we can have a backdook that exposes journal service for testing?

**Comment by CatsJuice** [comment]
> I think we need to add tests for the duplicated journal logic. However how to add duplicated journal is a question ... Perhaps we can have a backdook that exposes journal service for testing?

Yes, I am adding some basic test cases. Now we can turn regular page into journal and specify specific dates, so conflicts can be easily triggered.

File: packages/frontend/core/src/modules/journal/store/journal.ts (hunk 1 above)
Comment (pengx17) [reply packages/frontend/core/src/modules/journal/store/journal.ts]: I think a well designed `computed` should be able to batch the updates

**Review by EYHN — APPROVED** [review]

**Pull request merged**